src/ruby/puppet-server-lib/puppet/server/http_client.rb

```diff
@@ -35,8 +34,8 @@ def initialize(server, port, options = {})
 
     @server = server
     @port = port
-    @use_ssl = options[:use_ssl]
-    @protocol = @use_ssl ? "https" : "http"
+    self.class.use_ssl = options[:use_ssl]
```

Comment:
What would happen if the first request made through the client were for http (use_ssl = false) but at least one subsequent one were for https (use_ssl = true)?  Could we end up initializing a client without configuring an SSLContext and try to use that for the https calls?